### ISSUE TITLE
Surprising error note "this delimiter might not be properly closed"

### ISSUE THREAD

**Issue by arielb1** [rust-lang/rust#56834]
## STR

```Rust
fn main() {
    S { x: 4,
        y: 5 };
}
fn foo() {
```

## Expected Result

An error message pointing  to `foo`

## Actual Result

```
   Compiling playground v0.0.1 (/playground)
error: this file contains an un-closed delimiter
 --> src/main.rs:5:12
  |
2 |     S { x: 4,
  |       - this delimiter might not be properly closed...
3 |         y: 5 };
  |              - ...as it matches this but it has different indentation
4 | }
5 | fn foo() {
  |          - ^
  |          |
  |          un-closed delimiter
```

The heuristic feels incorrect to me - the delimiter on `S` looks perfectly well-closed.

**Comment by arielb1** [comment]
One way to improve the heuristic would be to not treat delimiters where the closing delimiter is more indented than the opening delimiter as suspicious. Are there any false negatives that this causes?

**Comment by arielb1** [comment]
cc @estebank - this lint was introduced in #53949 & #54092.

**Comment by iamgabrielsoft** [comment]
I had the same issue, try closing all empty braces from your for loop, if statement and function statement. 

very useful for me...
